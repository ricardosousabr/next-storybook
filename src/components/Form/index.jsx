import Button from "../Button";

export default function Form() {
  function handleForm(e) {
    e.preventDefault();

    console.log("Test");
  }
  return (
    <>
      <div>
        <form onClick={handleForm}>
          <input type="text" />
          <Button />
        </form>
      </div>
    </>
  );
}
