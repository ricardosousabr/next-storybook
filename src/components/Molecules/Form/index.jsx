import Button from "../../Atoms/Button/index";

export default function Form() {
  function handleForm(e) {
    e.preventDefault();

    console.log("Test");
  }
  return (
    <>
      <div>
        <form onClick={handleForm}>
          <input type="text" />
          <Button type="button" onClick={() => console.log("Test")}>Click</Button>
        </form>
      </div>
    </>
  );
}
