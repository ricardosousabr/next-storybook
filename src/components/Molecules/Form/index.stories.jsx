import Form from "./index.jsx";

export default {
  title: "Components/Form",
  Component: Form,
};

export const HomePage = () => <Form />;
