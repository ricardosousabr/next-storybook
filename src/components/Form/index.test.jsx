import renderer from "react-test-renderer";
import Form from "./index";

test("render form", () => {
  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
