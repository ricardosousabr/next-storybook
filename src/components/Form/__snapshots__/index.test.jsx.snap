// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render form 1`] = `
<div>
  <form
    onClick={[Function]}
  >
    <input
      type="text"
    />
    <div
      className="styles__BoxButton-sc-1f4hn43-0 gbLRBs"
    >
      <button
        className="styles__ButtonAqua-sc-1f4hn43-1 jkWVxD"
      />
    </div>
  </form>
</div>
`;
